### PR TITLE
Installation documentation 

### PR DESCRIPTION
**Pull request by jan-janssen**
I rewrote the installation documentation to cover the basic setup of a workstation up to the installation of a HPC cluster. I would be grateful if some of you can take the time to take a deeper look at this: 

* @ashtonmv  and @liamhuber maybe you can briefly skim through the document to check for spelling and german smell. 
* @niklassiemer and @muh-hassani as the two of you just joined the department maybe you can try to configure your local system to submit calculation to our cluster via SSH and copy the calculation files. I guess it is part of your projects to configure remote clusters at some points so you are the perfect candidates to test these instructions. 
* @sudarsan-surendralal as a windows user maybe you can add a few lines how to setup the windows subsystem for linux and what are special flags users should take care of when installing pyiron on Windows. 

The documentation is available on readthedocs as a separate branch: 
https://pyiron.readthedocs.io/en/installdocumentation/source/installation.html

### CODE REVIEW

**Comment by ashtonmv** [comment]
This is awesome @jan-janssen :tada: I can tell a lot of thought went into this and it's very well-written.

Feel free to reject any/all of the changes in my commit. I mostly just sprinkled around a few oxford commas or similar cosmetic suggestions. My hope was that a commit would make it easier to review the changes, and then the real expert @liamhuber can just start from here and clean up anything I might have missed. But feel free to also just revert the whole thing.

I had a few additional thoughts/questions:

- It's pretty long, and at some point I started asking myself if the content wouldn't fit better into several bite-sized rst files.
- On a related note, I wonder if there are any screenshots, etc. we could build in here to break up the ocean of text.
- Can you check that the line I added for conda-installing lammps on linux is correct?

```sh
    # serial + parallel, for linux systems
    conda install -c conda-forge lammps

    # only serial (no python bindings), for native windows
    conda install -c conda-forge -c pyiron lammps
```

- Should this line say "*pyiron* can’t delete files outside this directory?"

```rst
In particular pyiron users can not delete files outside this directory
```

**Comment by niklassiemer** [comment]
It is a great overview and I have learned quite a bit in configuration of pyiron!
I have a few questions/remarks:
The .pyiron on the HPC system is the one from the user, isn't it? I edited my .pyiron config on the garching cluster, downloaded the resources and edited the queues directory, however, so far I am still running locally (or not at all, since I do not have specific versions installed). Did I understand something wrong or do I have to specify something in addition?
In `Install from Source` we refer to pyiron's environment.yml file, which is somewhat hidden in the .ci-support folder. Would it not be nicer to expose this to the user?

**Comment by jan-janssen** [comment]
Thanks @ashtonmv for all the feedback, I definitely have a more detailed look over the course of the week. 

To address the questions:
- On readthedocs the different sections become separate headlines but I am open to having multiple files as well https://pyiron.readthedocs.io/en/installdocumentation/source/installation.html 
- Screenshot would definitely help, so far I was focusing on getting all the details included, as most of the configuration is still commandline based I am not so sure which screenshots to add but we could have a gif to demonstrate certain steps. Still that might already be another pull request. 
- the Linux line is correct in addition it also works for Mac OS X.
- Yes I agree. 

@niklassiemer thanks for trying it, maybe you can open a separate issue with your local .pyiron configuration, the one on the cluster and the corresponding resource directories. Once everything is configured you should be able to submit to the remote Queue by setting the server object, like it is illustrated on the website.

**Comment by jan-janssen** [comment]
I updated the text with Michaels recommendation, so unless there is more feedback I would merge this version to continue with the numfocus application and then we can still update the documentation with screenshots in a separate pull request.

**Review by sudarsan-surendralal — APPROVED** [review]
This is very well written with only a minor suggestion from my side!

Regarding windows installation: It's really been a long time since I used the windows installation (I use Windows Subsystem for Linux for running pyiron locally) so I'm not a 100% sure about it anymore. I guess with conda the installation is the same except that you can't install Sphinx/GPAW.

File: docs/source/installation.rst [hunk 1]
```diff
@@ -3,233 +3,498 @@
 ============
 Installation
 ============
+******************
+Conda Installation
+******************
+The recommended way to install pyiron is via the conda package manager in a Linux environment. So if you are using Windows we recommend using the `Windows subsystem for Linux <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`_ to install pyiron and if you are on macOS X we recommend using a `virtual machine/ virtual box <https://www.virtualbox.org>`_. Native installations on both Windows and macOS X are possible but are restricted to molecular dynamics calculations with interatomic potentials and do not support density functional theory(DFT) codes. We collaborate with the open-source community at `conda-forge <https://conda-forge.org>`_ to not only provide the pyiron package via their community channel, but also executables for compatible simulation codes like `GPAW <https://wiki.fysik.dtu.dk/gpaw/>`_, `LAMMPS <https://lammps.sandia.gov>`_ and `S/PHI/nX <https://sxrepo.mpie.de>`_ and their parameter files like pseudopotentials and interatomic potentials. To get started you can install pyiron using:
```
[the remark this answers is not in the document]
Comment (jan-janssen) [reply docs/source/installation.rst]: Done

**Pull request merged**